lib/pay/paddle/webhooks/subscription_payment_succeeded.rb

```diff
@@ -0,0 +1,41 @@
+module Pay
+  module Paddle
+    module Webhooks
+      class SubscriptionPaymentSucceeded
+
+        def initialize(data)
+          billable = Pay.find_billable(processor: :paddle, processor_id: data["user_id"])
+          return unless billable.present?
+          return if billable.charges.where(processor_id: data["subscription_payment_id"]).any?
+
+          charge = create_charge(billable, data)
+          notify_user(billable, charge)
+          charge
+        end
+
+        def create_charge(user, data)
+          charge = user.charges.find_or_initialize_by(
+            processor: :paddle,
+            processor_id: data["subscription_payment_id"]
+          )
+
+          charge.update(
+            amount: Integer(data["sale_gross"].to_f * 100),
+            card_type: data["payment_method"],
+            receipt_url: data["receipt_url"],
+            created_at: DateTime.parse(data["event_time"])
+          )
```

Comment:
Would be cool if we could store all the information from the webhook somehow. In postgres I'd create a jsonb column for that, but I'm not sure what would be an universal method for that.

Reply:
That's a great idea! Having a json column both on the subscription and the charges table would allow payment provider specific information to be saved easily. I know that @excid3 used a json column for storing params in the [noticed](https://github.com/excid3/noticed) gem. Maybe that would fit in here as well?

Reply:
JSON column would be nice, although compatibility is tough. I'm serializing and using text for sqlite, json for mysql, and jsonb for postgres in noticed and serializing with the ActiveJob coder. That works pretty well, but not queryable in all cases. I think that's probably okay. We can open up a separate issue for that.